migration/migration.go

```diff
@@ -19,20 +21,38 @@ import (
 	"github.com/sourcegraph/conc/pool"
 )
 
+type schemaMetadata struct {
+	Version   uint64
+	Done      bool
+	LastState []byte
+}
+
+func (schema *schemaMetadata) fromBytes(data []byte) error {
+	return json.Unmarshal(data, schema)
```

Comment:
Might be better to use `cbor` for consistency. We can discuss the more important parts of this PR before worrying about that, though.